src/components/MoneyReportHeader.tsx

```diff
@@ -470,6 +472,9 @@ function MoneyReportHeader({
         if (isPaidAnimationRunning) {
             return CONST.REPORT.PRIMARY_ACTIONS.PAY;
         }
+        if (isSubmittingAnimationRunning) {
+            return CONST.REPORT.PRIMARY_ACTIONS.SUBMIT;
+        }
```

Comment:
I think this logic, along with `isPaidAnimationRunning` and `!moneyRequestReport` should live in `getReportPrimaryAction` so that `getReportPrimaryAction` always returns the correct action no matter where it's being used

Reply:
Let's work on this one though